src/main.rs

```diff
@@ -310,13 +321,11 @@ async fn run_test(test_args: TestArgs) -> anyhow::Result<i32> {
         run_result.exit_code
     };
 
-    let upload_exit_code = run_upload(upload_args).await.unwrap_or(EXIT_SUCCESS);
-    // use the upload exit code if the command exit code is exit_success
-    if exit_code == EXIT_SUCCESS {
-        Ok(upload_exit_code)
-    } else {
-        Ok(exit_code)
+    let upload_exit_code = run_upload(upload_args).await.unwrap_or(EXIT_FAILURE);
```

Comment:
`unwrap` will lose context of an error which we want to log. I would use match for each case instead. Something like this

```rust
match run_upload(...)
{
    Ok(EXIT_SUCCESS) => EXIT_SUCCESS,
    Ok(code)         => { log:error("...");      EXIT_FAILURE },
    Err(err).        => { log:error("...", err); EXIT_FAILURE },
}
```

Reply:
Or if we don't need exit code:

```rust
match run_upload(...)
{
    Ok(EXIT_SUCCESS) => (),
    Ok(code)         => { log:error("...", code); },
    Err(err).        => { log:error("...", err);  },
}
```